File: app/src/jvx/java/art/coded/wireframe/view/fragment/CustomFragment.java
package art.coded.wireframe.view.fragment;

import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;

import art.coded.wireframe.databinding.FragmentCustomBinding;
import art.coded.wireframe.view.custom.CustomView;
import art.coded.wireframe.viewmodel.CustomViewModel;

public class CustomFragment extends Fragment {

    private static final String LOG_TAG = CustomFragment.class.getSimpleName();

    private FragmentCustomBinding binding;

    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container, Bundle savedInstanceState) {
        CustomViewModel customViewModel =
                new ViewModelProvider(this).get(CustomViewModel.class);

        binding = FragmentCustomBinding.inflate(inflater, container, false);
        View root = binding.getRoot();

        final TextView textView = binding.textCustom;
        binding.customView.setOnClickListener(view -> {
            int labelRes = ((CustomView) view).getLevelLabelRes();
            String label = getString(labelRes);
            customViewModel.setText(label);
        });
        customViewModel.getText().observe(getViewLifecycleOwner(), textView::setText);

        return root;
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }
}

File: app/src/ktx/java/art/coded/wireframe/view/fragment/CustomFragment.kt
package art.coded.wireframe.view.fragment

import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import androidx.lifecycle.ViewModelProvider
import art.coded.wireframe.viewmodel.CustomViewModel
import android.view.View
import androidx.fragment.app.Fragment
import art.coded.wireframe.databinding.FragmentCustomBinding

private val LOG_TAG = CustomFragment::class.java.simpleName

class CustomFragment : Fragment() {
    private var binding: FragmentCustomBinding? = null
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?, savedInstanceState: Bundle?
    ): View {
        val customViewModel = ViewModelProvider(this).get(CustomViewModel::class.java)
        binding = FragmentCustomBinding.inflate(inflater, container, false)
        val root: View = binding!!.root
        val textView = binding!!.textCustom
        val customView = binding!!.customView
        customView.setOnClickListener {
            val labelRes = customView.levelLabelRes
            val label = getString(labelRes)
            customViewModel.text.value = label
        }
        customViewModel.text
            .observe(viewLifecycleOwner) { o: String? -> textView.text = o }
        return root
    }

    override fun onDestroyView() {
        super.onDestroyView()
        binding = null
    }
}